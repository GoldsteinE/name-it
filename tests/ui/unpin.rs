use name_it::name_it;

#[name_it(Foo)]
async fn foo() {
}

impl Unpin for _foo_impl::Foo<'_> {}

fn assert_unpin(_: impl Unpin) {}

fn main() {
    assert_unpin(foo());
}
