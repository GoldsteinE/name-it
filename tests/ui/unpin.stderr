error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unpin.rs:12:18
   |
12 |     assert_unpin(foo());
   |     ------------ ^^^^^ within `Named<_foo_impl::Foo<'_>>`, the trait `Unpin` is not implemented for `PhantomPinned`
   |     |
   |     required by a bound introduced by this call
   |
   = note: consider using `Box::pin`
   = note: required because it appears within the type `Named<_foo_impl::Foo<'_>>`
note: required by a bound in `assert_unpin`
  --> tests/ui/unpin.rs:9:25
   |
9  | fn assert_unpin(_: impl Unpin) {}
   |                         ^^^^^ required by this bound in `assert_unpin`
